sycl/test-e2e/DeviceLib/built-ins/offload-prec-div-sqrt.cpp

```diff
@@ -0,0 +1,67 @@
+// RUN: %{build} -foffload-fp32-prec-div -foffload-fp32-prec-sqrt -o %t.out
+// RUN: %{run} %t.out
+
+// Test if div and sqrt become precise from IEEE-754 perspective when
+// -foffload-fp32-prec-div -foffload-fp32-prec-sqrt are passed.
+
+#include <cmath>
+#include <sycl/detail/core.hpp>
+#include <sycl/usm.hpp>
+
+constexpr float value = 560.0f;
+constexpr float divider = 280.0f;
+
+// Reference
+// https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/util/accuracy.h
+template <typename T> T get_ulp_std(T x) {
+  const T inf = std::numeric_limits<T>::infinity();
+  const T negative = std::fabs(std::nextafter(x, -inf) - x);
+  const T positive = std::fabs(std::nextafter(x, inf) - x);
+  return std::fmin(negative, positive);
+}
+
+template <typename T> int ulp_difference(const T &lhs, const T &rhs) {
```

Comment:
Why does it return `int`?

Reply:
The intention was to remove ulps diff as an integer, but implicit cast was undesired here (and actually was wrong). I've simplified the function.